test/javascript/components/concept-map/ConceptMap.test.tsx

```diff
@@ -8,78 +8,82 @@ import '@testing-library/jest-dom/extend-expect'
+import { IConceptMap } from '../../../../app/javascript/components/concept-map/concept-map-types'
 
 describe('<ConceptMap />', () => {
-  test('renders empty component', () => {
-    const { container } = render(
-      <ConceptMap
-        concepts={[]}
-        levels={[[]]}
-        connections={[]}
-        status={{}}
-        exercise_counts={{}}
-      />
-    )
+  test('renders empty component', async () => {
+    const config: IConceptMap = {
+      concepts: [],
+      levels: [[]],
+      connections: [],
+      status: {},
+      exercise_counts: {},
+    }
+
+    const { container } = await waitForConceptMapReady(config)
     const map = container.querySelector('.c-concepts-map')
     expect(map).not.toBeNull()
   })
 
   test('renders single incomplete concept', async () => {
     const testConcept = concept('test')
 
-    render(
-      <ConceptMap
-        concepts={[testConcept]}
-        levels={[[testConcept.slug]]}
-        connections={[]}
-        status={{
-          test: 'unavailable',
-        }}
-        exercise_counts={{
-          test: {
-            exercises: 1,
-            exercises_completed: 0,
-          },
-        }}
-      />
-    )
-
-    await waitFor(() => {
-      expect(screen.getByText('Test')).toBeInTheDocument()
-      expect(
-        screen.queryByTitle('You have mastered this concept')
-      ).not.toBeInTheDocument()
-    })
+    const config: IConceptMap = {
+      concepts: [testConcept],
+      levels: [[testConcept.slug]],
+      connections: [],
+      status: { test: 'unavailable' },
+      exercise_counts: { test: { exercises: 1, exercises_completed: 0 } },
+    }
+
+    await waitForConceptMapReady(config)
+
+    expect(
+      screen.queryByTitle('You have mastered this concept')
+    ).not.toBeInTheDocument()
   })
 
   test('renders single completed concept', async () => {
     const testConcept = concept('test')
-    render(
-      <ConceptMap
-        concepts={[testConcept]}
-        levels={[[testConcept.slug]]}
-        connections={[]}
-        status={{
-          test: 'completed',
-        }}
-        exercise_counts={{
-          test: {
-            exercises: 1,
-            exercises_completed: 1,
-          },
-        }}
-      />
-    )
-
-    await waitFor(() => {
-      expect(screen.getByText('Test')).toBeInTheDocument()
-      expect(
-        screen.getByTitle('You have mastered this concept')
-      ).toBeInTheDocument()
-    })
+
+    const config: IConceptMap = {
+      concepts: [testConcept],
+      levels: [[testConcept.slug]],
+      connections: [],
+      status: { test: 'unavailable' },
+      exercise_counts: { test: { exercises: 1, exercises_completed: 1 } },
+    }
+
+    await waitForConceptMapReady(config)
+
+    expect(
+      screen.queryByTitle('You have mastered this concept')
+    ).toBeInTheDocument()
   })
 })
 
+const waitForConceptMapReady = async (config: IConceptMap) => {
+  const renderResult = render(
```

Comment:
Is this renderResult necessary? I'd either expect this to be called `renderConceptMap` or take in a rendered concept map and replace `screen.getByText` to use the `render.getByText`

Reply:
I had just named it after the return type from `render`. I can split them up for versatility, sure.

Reply:
Splitting up might make it easier for others to read; it's weird that the result of this function is never used. You often want to scope your `getByXXX` to the render result, so it can't find something somewhere else on the page! (Not really applicable to your minimal example, but will be in compound/larger tests).